lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -877,6 +934,55 @@ def configure_cdrom(self, vm_obj):
                 self.change_detected = True
                 self.configspec.deviceChange.append(cdrom_spec)
 
+    def configure_floppy(self, vm_obj):
+        # Configure the VM floppy
+        if self.params.get("floppy", None):
+            if "type" not in self.params["floppy"] or self.params["floppy"]["type"] not in ["none", "client", "flp"]:
+                self.module.fail_json(msg="floppy.type is mandatory")
+            if self.params["floppy"]["type"] == "flp" and ("flp_path" not in self.params["floppy"] or not self.params["floppy"]["flp_path"]):
+                self.module.fail_json(msg="floppy.flp_path is mandatory in case floppy.type is flp")
+
+            if vm_obj and vm_obj.config.template:
+                # Changing floppy settings on a template is not supported
+                return
```

Comment:
Should there be an error to the user here instead of silently doing nothing?

Reply:
Yes, agreed